lib/princeton_marc.rb

```diff
@@ -149,6 +149,51 @@ def process_names record
   names.uniq
 end
 
+##
+# Get hash of authors grouped by role
+# @return [Hash]
+def process_author_roles record
+  author_roles = {
+    'TRL' => 'translators',
+    'EDT' => 'editors',
+    'COM' => 'compilers'
+  }
+  names = {}
+  names['secondary_authors'] = []
+  names['translators'] = []
+  names['editors'] = []
+  names['compilers'] = []
+
+  Traject::MarcExtractor.cached('100a:110a:111a:700a:710a:711a').collect_matching_lines(record) do |field, spec, extractor|
+    name = extractor.collect_subfields(field, spec).first
+    unless name.nil?
+      name = Traject::Macros::Marc21.trim_punctuation(name)
+
+      # If name is from 1xx field, it is the primary author.
+      if /1../.match(field.tag)
+        names['primary_author'] = name
+      else
+        field.subfields.each do |s_field|
+
+          # If field has relator code (subfield 4).
+          if s_field.code == '4'
```

Comment:
When a 1xx/7xx name has an `$e` we use that value as the label in Blacklight. There are multiple records that don't have a standardized term in `$e` (like 'ill.' instead of 'illustrator), which makes our decision to include the relator as the label a bit 'controversial'